src/Symfony/Component/Security/Core/Authorization/AccessDecisionStrategyInterface.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Symfony\Component\Security\Core\Authorization;
+
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+
+/**
+ * AccessDecisionStrategyInterface contains the strategy to make access decissions.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+interface AccessDecisionStrategyInterface
+{
+    /**
+     * Configures the voters.
+     *
+     * @param VoterInterface[] $voters An array of VoterInterface instances
+     */
+    public function setVoters(array $voters);
+
+    /**
+     * Set whether to grant access if all voters abstained or not.
+     *
+     * @param bool $allowIfAllAbstainDecisions
+     */
+    public function setAllowIfAllAbstainDecisions($allowIfAllAbstainDecisions);
+
+    /**
+     * Set whether to grant access if result are equals.
+     *
+     * @param bool $allowIfEqualGrantedDeniedDecisions
+     */
+    public function setAllowIfEqualGrantedDeniedDecisions($allowIfEqualGrantedDeniedDecisions);
```

Comment:
these setters don't make sense for all strategies. They only make sense for the 3 existing strategies